File: lessons/day_06/netcore/src/GraphQLSample/Models/Input/InputCandidateCreate.cs
using GraphQLNetCore.Models.Enums;
using System.Collections.Generic;

namespace GraphQLNetCore.Models.Input
{
   public class InputCandidateCreate
   {
      public string Name { get; set; }
      public string Surname { get; set; }
      public string Email { get; set; }
      public int Age { get; set; }
      public EyeColor? EyeColor { get; set; }
      public List<int> Friends { get; set; }
      public List<int> Skills { get; set; }
      public int? FavSkill { get; set; }
      public Role TargetRole { get; set; }
      public Grade TargetGrade { get; set; }

      internal Candidate ToPerson()
      {
         return new Candidate() {
            Age = Age,
            Email = Email,
            EyeColor = EyeColor,
            FavSkillId = FavSkill,
            Name = Name,
            Surname = Surname,
            TargetRole = TargetRole,
            TargetGrade = TargetGrade
         };
      }
   }
}

File: lessons/day_07/java/src/main/java/com/example/DemoGraphQL/input/InputCandidateCreate.java
package com.example.DemoGraphQL.input;

import com.example.DemoGraphQL.model.EyeColor;
import com.example.DemoGraphQL.model.Grade;
import com.example.DemoGraphQL.model.Role;

import java.util.List;

public class InputCandidateCreate {
    private String name;
    private String surname;
    private String email;
    private Integer age;
    private EyeColor eyeColor;
    private Long favSkill;
    private List<Long> friends;
    private List<Long> skills;
    private Role targetRole;
    private Grade targetGrade;

    public InputCandidateCreate() {
    }

    public InputCandidateCreate(String name, String surname, String email, Integer age, EyeColor eyeColor, Long favSkill, List<Long> friends, List<Long> skills, Role targetRole, Grade targetGrade) {
        this.name = name;
        this.surname = surname;
        this.email = email;
        this.age = age;
        this.eyeColor = eyeColor;
        this.favSkill = favSkill;
        this.friends = friends;
        this.skills = skills;
        this.targetRole = targetRole;
        this.targetGrade = targetGrade;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public EyeColor getEyeColor() {
        return eyeColor;
    }

    public void setEyeColor(EyeColor eyeColor) {
        this.eyeColor = eyeColor;
    }

    public Long getFavSkill() {
        return favSkill;
    }

    public void setFavSkill(Long favSkill) {
        this.favSkill = favSkill;
    }

    public List<Long> getFriends() {
        return friends;
    }

    public void setFriends(List<Long> friends) {
        this.friends = friends;
    }

    public List<Long> getSkills() {
        return skills;
    }

    public void setSkills(List<Long> skills) {
        this.skills = skills;
    }

    public Role getTargetRole() {
        return targetRole;
    }

    public void setTargetRole(Role targetRole) {
        this.targetRole = targetRole;
    }

    public Grade getTargetGrade() {
        return targetGrade;
    }

    public void setTargetGrade(Grade targetGrade) {
        this.targetGrade = targetGrade;
    }
}
